connectors/es/sink.py

```diff
@@ -391,6 +435,12 @@ async def _run(self):
                         ),
                         name=f"Elasticsearch Sink: _bulk batch #{batch_num}",
                     )
+
+                    if force_flush:
+                        # Release whoever was waiting for flush to happen
+                        self._logger.debug("Releasing the flush operation awaiter")
+                        doc.trigger()
```

Comment:
this call to `doc.trigger()` is happening outside the block where you checked `isinstance(doc, ForceFlushSignal)`. Which is safe because you're using `force_flush` to ensure the same condition is true. But I wonder if this is just an indicator that we need a `def flush()` function that can be called from various places, so that the code can look like:
```
if isinstance(doc, ForceFlushSignal):
  flush("from signal")
else:
  add_to_batch(doc)
  if batch_full():
    flush("from capacity")
```

Reply:
Might be a good idea indeed, for readability and future maintenance!